### ISSUE TITLE
Add Stochastic Weight Averaging (SWA) Optimizer

### ISSUE THREAD

**Issue by shreyashpatodia** [tensorflow/addons#592]
__Describe the feature and the current behavior/state.__

SWA has been shown to be a simple method to improve model performance when used as a wrapper around other optimizers and averaging results from different points of trajectory of the inner optimizer.

__Relevant information__

- Are you willing to contribute it (yes/no): yes, I have an implementation that I should be able to port to the repository fairly easily. The implementation looks pretty similar to the implementation for the Lookahead Optimizer that was recently added to addons.
- Are you willing to maintain it going forward? (yes/no): yes
- Is there a relevant academic paper? (if so, where):  yes, [Averaging Weights Leads to Wider Optima and Better Generalization](https://arxiv.org/abs/1803.05407)
- Is there already an implementation in another framework? (if so, where): yes, [Pytorch](https://github.com/pytorch/contrib/tree/master/torchcontrib/optim) and [fast.ai](https://github.com/fastai/fastai/blob/master/old/fastai/swa.py) (this is in old right now because they seem to be prepping for a merge to the new version).
- Was it part of tf.contrib? (if so, where): No 

__Which API type would this fall under (layer, metric, optimizer, etc.)?__

This would fall under the optimizer API.

__Who will benefit with this feature?__

SWA works like any other optimizer so it can be a good addition to standard Adam, SGD or RMSProp to help squeeze out a little bit of extra performance.

**Comment by simon-larsson** [comment]
I have made a callback implementation of SWA here: https://github.com/simon-larsson/keras-swa. Feel free to use if it helps.

**Comment by shreyashpatodia** [comment]
I'll make a PR for this soon.
